src/ui/public/vis/vis.js

```diff
@@ -8,184 +8,183 @@
+
+    setUiState(uiState) {
+      if (uiState instanceof PersistedState) {
+        this.__uiState = uiState;
       }
-    } else {
-      this.type = type;
-    }
-
-    this.listeners = _.assign({}, state.listeners, this.type.listeners);
-    this.params = _.defaults({},
-      _.cloneDeep(state.params || {}),
-      _.cloneDeep(this.type.params.defaults || {})
-    );
-
-    this.aggs = new AggConfigs(this, state.aggs);
-  };
-
-  Vis.prototype.getStateInternal = function (includeDisabled) {
-    return {
-      title: this.title,
-      type: this.type.name,
-      params: this.params,
-      aggs: this.aggs
-        .filter(agg => includeDisabled || agg.enabled)
-        .map(agg => agg.toJSON())
-        .filter(Boolean),
-      listeners: this.listeners
-    };
-  };
-
-  Vis.prototype.getEnabledState = function () {
-    return this.getStateInternal(false);
-  };
-
-  Vis.prototype.getState = function () {
-    return this.getStateInternal(true);
-  };
-
-  Vis.prototype.createEditableVis = function () {
-    return this._editableVis || (this._editableVis = this.clone());
-  };
-
-  Vis.prototype.getEditableVis = function () {
-    return this._editableVis || undefined;
-  };
-
-  Vis.prototype.clone = function () {
-    const uiJson = this.hasUiState() ? this.getUiState().toJSON() : {};
-    return new Vis(this.indexPattern, this.getState(), uiJson);
-  };
-
-  Vis.prototype.requesting = function () {
-    // Invoke requesting() on each agg. Aggs is an instance of AggConfigs.
-    _.invoke(this.aggs.getRequestAggs(), 'requesting');
-  };
-
-  Vis.prototype.isHierarchical = function () {
-    if (_.isFunction(this.type.hierarchicalData)) {
-      return !!this.type.hierarchicalData(this);
-    } else {
-      return !!this.type.hierarchicalData;
-    }
-  };
-
-  Vis.prototype.hasSchemaAgg = function (schemaName, aggTypeName) {
-    const aggs = this.aggs.bySchemaName[schemaName] || [];
-    return aggs.some(function (agg) {
-      if (!agg.type || !agg.type.name) return false;
-      return agg.type.name === aggTypeName;
-    });
-  };
-
-  Vis.prototype.hasUiState = function () {
-    return !!this.__uiState;
-  };
-  Vis.prototype.setUiState = function (uiState) {
-    if (uiState instanceof PersistedState) {
-      this.__uiState = uiState;
-    }
-  };
-  Vis.prototype.getUiState = function () {
-    return this.__uiState;
-  };
-
-  Vis.prototype.implementsRenderComplete = function () {
-    return this.type.implementsRenderComplete;
-  };
-
-  /**
-   * Currently this is only used to extract map-specific information
-   * (e.g. mapZoom, mapCenter).
-   */
-  Vis.prototype.uiStateVal = function (key, val) {
-    if (this.hasUiState()) {
-      if (_.isUndefined(val)) {
-        return this.__uiState.get(key);
+    }
+
+    getUiState() {
+      return this.__uiState;
+    }
+
+    /**
+     * Currently this is only used to extract map-specific information
+     * (e.g. mapZoom, mapCenter).
+     */
+    uiStateVal(key, val) {
```

Comment:
no idea, this is still there from the old code and the comment mentions it only being used it maps ...